### PR TITLE
Standards

### PR DESCRIPTION
**Pull request by adamdbrw**
Coding standards - see commits.
First batch of changes based on https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md.
Second part will come once we figure out how to apply linters to Gems (solo)

### CODE REVIEW

File: Gems/ROS2/Code/Source/Clock/SimulationClock.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#include "SimulationClock.h"
```
Comment (mdrwiega) [inline Gems/ROS2/Code/Source/Clock/SimulationClock.cpp:1]: Maybe it should also contain the copyright note?

File: Gems/ROS2/Code/ros2_files.cmake [hunk 2]
```diff
@@ -1,14 +1,14 @@
 
 set(FILES
     Include/ROS2/ROS2Bus.h
-    Source/LidarRaycaster.cpp
-    Source/LidarRaycaster.h
-    Source/LidarTemplate.h
-    Source/ROS2LidarSensorComponent.cpp
-    Source/ROS2LidarSensorComponent.h
+        Source/Lidar/LidarRaycaster.cpp
```
Comment (mdrwiega) [inline Gems/ROS2/Code/ros2_files.cmake:4]: Is this formatting correct?
Looks like the files paths are not aligned.

File: Gems/ROS2/Code/Source/Clock/SimulationClock.cpp (hunk 1 above)
Comment (adamdbrw) [reply Gems/ROS2/Code/Source/Clock/SimulationClock.cpp:8]: Missed this one, thank you!

File: Gems/ROS2/Code/ros2_files.cmake (hunk 2 above)
Comment (adamdbrw) [reply Gems/ROS2/Code/ros2_files.cmake]: My Clion needs to be tamed (an effect of auto refactor when dropping in folder). Fixed

**Review by mdrwiega — APPROVED** [review]

**Pull request merged**